File: learning-analytics/learning-model/src/main/java/com/example/learning/model/entity/KnowledgePoint.java
package com.example.learning.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * KnowledgePoint 实体类
 * 对应数据库表: knowledge_point
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("knowledge_point")
public class KnowledgePoint {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long tenantId;

    private Long parentId;

    private Long subjectId;

    private String kpCode;

    private String kpName;

    private Integer level;

    private Integer sortOrder;

    private Integer status;

    @TableField("created_at")
    private LocalDateTime createdAt;
}

File: learning-analytics-web/src/api/knowledge-point.ts
import { get, post, put, del } from './request'

/** 知识点信息 */
export interface KnowledgePointItem {
  id: number
  tenantId: number
  parentId: number
  subjectId: number
  kpCode: string
  kpName: string
  level: number
  sortOrder: number
  status: number
  createdAt: string
  children?: KnowledgePointItem[]
}

/** 创建/编辑知识点参数 */
export interface KnowledgePointFormData {
  id?: number
  parentId?: number
  subjectId: number
  kpCode: string
  kpName: string
  level?: number
  sortOrder?: number
  status?: number
}

/** 获取知识点树形列表 */
export function getKnowledgePointTree(subjectId: number) {
  return get<KnowledgePointItem[]>('/knowledge-points', { subjectId })
}

/** 创建知识点 */
export function createKnowledgePoint(data: KnowledgePointFormData) {
  return post<KnowledgePointItem>('/knowledge-points', data)
}

/** 更新知识点 */
export function updateKnowledgePoint(id: number, data: KnowledgePointFormData) {
  return put<KnowledgePointItem>(`/knowledge-points/${id}`, data)
}

/** 删除知识点 */
export function deleteKnowledgePoint(id: number) {
  return del(`/knowledge-points/${id}`)
}
